### DISCUSSION TITLE
How to preview PDFs with FileElement/MutlifileElement

### DISCUSSION THREAD

**Discussion by maxwellsmart84** [vueform/vueform#142] [Questions] [answered]
I am trying to get image previewing to work the gallery view when a user uploads a PDF.  

Is there a way to use a template to customize this so I can use an `embed` tag?  

Is there a way to do this with the built-in API that I am not aware of?  I notice that the preview is just an <`img src=<base64EncodedString>`.  Is there any work around for this?

Thanks in advance

**Comment by adamberecz** [comment, marked as answer]
You can create an [alternative view](https://vueform.com/docs/styles-and-layout#alternative-views) eg. `_pdf` for FilePreview component:

```js
// vueform.config.js

import FilePreview_pdf from './FilePreview_pdf.vue'

export default {
  // ...
  templates: {
    FilePreview_pdf,
  }
}
```

Then you can use it with:

```vue
<FileElement view="pdf" />
```

Here's a starting point for the structure that you can copy:
https://github.com/vueform/vueform/blob/main/themes/blank/templates/elements/partials/FilePreview_image.vue

Then you can fill classes from here (vueform theme):
https://github.com/vueform/vueform/blob/main/themes/vueform/templates/elements/partials/FilePreview_image.vue#L10

Or from here (tailwind theme):
https://github.com/vueform/vueform/blob/main/themes/tailwind/classes.js#L1694

**Reply by maxwellsmart84** [reply]
Thank you so much 👍
